.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,85 @@
+name: CI/CD Pipeline
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      - 'App/**'
+      - 'CourseAdmin/**'
+      - 'Server/**'
```

Comment:
I think the way the paths are set up is that both the frontend and backend tests are run regardless of if their respective projects were changed (for example, a pure CSS change would still trigger backend tests). This shouldn't be too much of a problem for now since we have few tests, but if we notice that there's too much compute time being used here (because Github free-tier does have quite stringent compute limits), we can optimize later.